cmd/kube-apiserver/app/server.go

```diff
@@ -145,6 +145,25 @@ func Run(s *options.APIServer) error {
 		glog.Fatalf("Failed to start kubelet client: %v", err)
 	}
 
+	if s.StorageConfig.DeserializationCacheSize == 0 {
+		// When size of cache is not explicitly set, estimate its size based on
+		// target memory usage.
+		glog.V(2).Infof("Initalizing deserialization cache size based on %dMB limit", s.TargetRAMMB)
+
+		// This is the heuristics that from memory capacity is trying to infer
```

Comment:
FYI - This is pretty-much copy from:
https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/cachesize/cachesize.go#L75